### PR TITLE
NOTIF-116 Introduce Bundles

### PR DESCRIPTION
**Pull request by pilhuhn**
Introduce Bundles as group of applications. 

Bundle->Applications is 1:n
The application now has a field bundle_id.
Application names are now only unique within a bundle.

The PR also adds some python code in helpers that one can use to quickly generate a Bundle+Application+EventType

### CODE REVIEW

File: src/main/java/com/redhat/cloud/notifications/db/BundleResources.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import reactor.core.publisher.Flux;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * Deal with Bundles.
+ * More or less stupid copy & paste & search & replace of
+ * code in BundleResources
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/db/BundleResources.java:17]: Did you do search & replace on this comment too? :grin:  (Was expecting to see `code in ApplicationResources`)

Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/notifications/db/BundleResources.java:17]: I guess I should change this a bit :)
I startet with c&p&s&r, but then also introduced that common base class (for deletes atm only), which could probably also be used elsewhere

**Comment by pilhuhn** [comment]
This currently fails with GH actions having issues to start Docker

```
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=postgres:latest, imagePullPolicy=DefaultPullPolicy())
69
Caused by: com.github.dockerjava.api.exception.NotFoundException: 
70
{"message":"No such image: testcontainersofficial/ryuk:0.3.0"}
```

It (still) exists on Dockerhub though: https://hub.docker.com/r/testcontainersofficial/ryuk/tags?page=1&ordering=last_updated

**Review by josejulio — APPROVED** [review]
Code looks good. Tests passes locally.

**Comment by pilhuhn** [comment]
Still the docker issue in GH Actions

File: src/main/java/com/redhat/cloud/notifications/models/Application.java [hunk 2]
```diff
@@ -31,6 +33,11 @@
     @JsonInclude(JsonInclude.Include.NON_NULL)
     private Set<EventType> eventTypes; // optional
 
+    @NotNull
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty("bundle_id")
+    private UUID bundleId;
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/models/Application.java:39]: This one is marked as non null and at least 1 API call doesn't return it. (It breaks a validation in the openapi.json)

Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/notifications/models/Application.java:39]: Good catch and thanks for the PR. I need to improve the tests.

**Pull request merged**